pkg/xds/bootstrap/generator.go

```diff
@@ -307,14 +348,27 @@ func (s SANSet) slice() []string {
 	return sans
 }
 
+func isNotCA(cert []byte) (bool, error) {
```

Comment:
I'd better change it to 
```go
func validateCA(cert []byte) error {
...
}
```
and if it's not CA then return `NotCA` straight away. 

It will simplify the usage of this method:
```go
func (b *bootstrapGenerator) validateCaCert(cert []byte, request types.BootstrapRequest) error {
	if request.DataplaneTokenPath != "" {
		// when using GoogleGRPC it is valid to put non-CA certificate therefore we should only verify this for EnvoyGRPC
		// EnvoyGRPC is used when DataplaneToken is passed via request as inline value, not as a file.
		return nil
	}

	return validateCA(cert)
}
```

And probably, in that case, we don't need a separate `validateCA` method, I guess its content can be inlined